pkg/monitor/cluster/cluster.go

```diff
@@ -91,14 +91,9 @@ func NewMonitor(log *logrus.Entry, restConfig *rest.Config, oc *api.OpenShiftClu
 		return nil, err
 	}
 
-	var hiveclientset client.Client
-	if hiveRestConfig != nil {
-		var err error
-		hiveclientset, err = client.New(hiveRestConfig, client.Options{})
-		if err != nil {
-			// TODO(hive): Update to fail once we have Hive everywhere in prod and dev
-			log.Error(err)
-		}
```

Comment:
nit, an alternative way is simply removing the outer "hiveRestConfig != nil" check, only keep the inner thing, like all other modules do:

	hiveclientset, err := client.New(hiveRestConfig, client.Options{})
	if err != nil {
		// TODO(hive): Update to fail once we have Hive everywhere in prod and dev
		log.Error(err)
	}


if have to keep "hiveRestConfig != nil" check, it's better to move it to the beginning of the func as other modules before this will crash first in that case, so the check here does not help.

Reply:
@yjst2012 Do you mean just removing 
```go
if hiveRestConfig == nil {
```
from pkg/monitor/cluster/cluster.go.NewMonitor() without any other change?

Reply:
yep, something like this will do the trick I believe, since the problem is introduced by "var hiveclientset hiveclient.Interface".

-       var hiveclientset hiveclient.Interface
-       if hiveRestConfig != nil {
-               var err error
-               hiveclientset, err = hiveclient.NewForConfig(hiveRestConfig)
-               if err != nil {
-                       // TODO(hive): Update to fail once we have Hive everywhere in prod and dev
-                       log.Error(err)
-               }
+       hiveclientset, err := hiveclient.NewForConfig(hiveRestConfig)
+       if err != nil {
+               // TODO(hive): Update to fail once we have Hive everywhere in prod and dev
+               log.Error(err)
        }

Reply:
Aha, I know the problem, cos I'm looking at a relatively old version that I'm working on. it still calling the old api 

`func NewForConfig(c *rest.Config) (*Clientset, error) {
`

which do not have the problem introduced :joy: